src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
+
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEAFULT_SIZE = 8;
+    private static Object[] stack;
```

Comment:
The 'stack' and 'size' fields should not be static. Making them static means they are shared across all instances of MateStack, which is not the intended behavior for a stack instance.